### ISSUE TITLE
null self.out in `if self.display: self.out.update(HTML(self.progress))`

### ISSUE THREAD

**Issue by 8enmann** [fastai/fastprogress#31]
```py
src = ItemLists(path, TextList(train_tok, path=path, processor=processor),
                TextList(valid_tok, path=path, processor=processor))
src = src.label_for_lm()
```
Produces this error:
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-11-f14467f55f2d> in <module>()
      4 src = ItemLists(path, TextList(train_tok, path=path, processor=processor),
      5                 TextList(valid_tok, path=path, processor=processor))
----> 6 src = src.label_for_lm()
      7 if test_tok is not None: src.add_test(TextList(test_tok, path=path))
      8 text_data = src.databunch()

/Users/ben/anaconda3/lib/python3.6/site-packages/fastai/data_block.py in _inner(*args, **kwargs)
    466             self.valid = fv(*args, from_item_lists=True, **kwargs)
    467             self.__class__ = LabelLists
--> 468             self.process()
    469             return self
    470         return _inner

/Users/ben/anaconda3/lib/python3.6/site-packages/fastai/data_block.py in process(self)
    520         "Process the inner datasets."
    521         xp,yp = self.get_processors()
--> 522         for ds,n in zip(self.lists, ['train','valid','test']): ds.process(xp, yp, name=n)
    523         #progress_bar clear the outputs so in some case warnings issued during processing disappear.
    524         for ds in self.lists:

/Users/ben/anaconda3/lib/python3.6/site-packages/fastai/data_block.py in process(self, xp, yp, name)
    697                     p.warns = []
    698                 self.x,self.y = self.x[~filt],self.y[~filt]
--> 699         self.x.process(xp)
    700         return self
    701 

/Users/ben/anaconda3/lib/python3.6/site-packages/fastai/data_block.py in process(self, processor)
     73         if processor is not None: self.processor = processor
     74         self.processor = listify(self.processor)
---> 75         for p in self.processor: p.process(self)
     76         return self
     77 

/Users/ben/anaconda3/lib/python3.6/site-packages/fastai/text/data.py in process(self, ds)
    290         ds.items = _join_texts(ds.items, self.mark_fields, self.include_bos, self.include_eos)
    291         tokens = []
--> 292         for i in progress_bar(range(0,len(ds),self.chunksize), leave=False):
    293             tokens += self.tokenizer.process_all(ds.items[i:i+self.chunksize])
    294         ds.items = tokens

/Users/ben/anaconda3/lib/python3.6/site-packages/fastprogress/fastprogress.py in __iter__(self)
     62     def __iter__(self):
     63         self.on_iter_begin()
---> 64         self.update(0)
     65         try:
     66             for i,o in enumerate(self._gen):

/Users/ben/anaconda3/lib/python3.6/site-packages/fastprogress/fastprogress.py in update(self, val)
     77             self.pred_t = 0
     78             self.last_v,self.wait_for = 0,1
---> 79             self.update_bar(0)
     80         elif val >= self.last_v + self.wait_for or val == self.total:
     81             cur_t = time()

/Users/ben/anaconda3/lib/python3.6/site-packages/fastprogress/fastprogress.py in update_bar(self, val)
     96             warn("Your generator is empty.")
     97             self.on_update(0, '100% [0/0]')
---> 98         else: self.on_update(val, f'{100 * val/self.total:.2f}% [{val}/{self.total} {elapsed_t}<{remaining_t}{end}]')
     99 
    100 

/Users/ben/anaconda3/lib/python3.6/site-packages/fastprogress/fastprogress.py in on_update(self, val, text, interrupted)
    167     def on_update(self, val, text, interrupted=False):
    168         self.progress = html_progress_bar(val, self.total, text, interrupted)
--> 169         if self.display: self.out.update(HTML(self.progress))
    170         elif self.parent is not None: self.parent.show()
    171 

AttributeError: 'NoneType' object has no attribute 'update'
```

**Comment by 8enmann** [comment]
Oops it's because I'm using py3.6. Fixed in 3.7. Would be nice to have a warning somewhere.

**Comment by sgugger** [comment]
Mmmm, that's weird there is nothing that uses python 3.7 code and I'm pretty sure I ran it with 3.6 without problems.
